stages/org.osbuild.firewall

```diff
@@ -62,6 +62,30 @@ SCHEMA = """
   "default_zone": {
     "description": "Set default zone for connections and interfaces where no zone has been selected.",
     "type": "string"
+  },
+  "sources": {
+    "description": "Bind the source to a zone",
+    "type": "array",
+    "items": {
+      "type": "object",
+      "additionalProperties": false,
+      "required": ["sources"],
+      "properties": {
+        "sources": {
+          "description": "A list of sources",
+          "type": "array",
+          "minItems": 1,
+          "items": {
+            "type": "string",
+            "description": "A source: [/<mask>]|<MAC>|ipset:<ipset>"
+          }
+        },
+        "zone": {
+          "description": "The zone to apply the source to",
+          "type": "string"
+        }
+      }
+    }
```

Comment:
This configuration allows specifying the same zone multiple times in the `sources` array. But that might be ok. There's no particularly nice way to do this in JSON, other than using patternProperties and making the key the zone-name.

Also note that a lot of firewall properties are based on the zone, so maybe we want to make this a high-level property to avoid duplicating it in the future? But then again, maybe we can restructure things once we have more options. Just putting this out here to keep in mind.

Reply:
How about we change it to this:

```json
"zones": [
    {
        "name": "<name0>",
        "sources": [
            "<source0>",
            "<source1>",
            ...
        ]
    },
    {
        "name": "<name1>",
        ...
    }
]
```

This avoids the duplicated `"sources": [{ "sources": [ ... ] }]` that we have in your proposed configuration. And it will allow us to easily add other zone-configuration in the future, like interface-bindings, forwarding, and policies.

The downside is that we currently do not allow creating zones, and this configuration looks like it does. However, I would argue that at one point we want that anyway, and then it seems like a logical extension. And we can always treat it as a `create if non-existant, amend otherwise`.

Does that sounds reasonable? Is there a downside compared to the current approach?

Reply:
this is completely reasonable, do you migrate the other schema to this tho? the other configuration for this stage apply to the default zone anyway, and passing an empty zone in your new schema would still apply it to the default zone right?

Reply:
@dvdhrm , I'm taking over the rest of this PR. 
I'm changing the schema to have the `zones` as the top-level property instead of `sources` as you both discussed. I will mark both `name` and `sources` of the `zones`' items as required, then it will continue applying the sources to the default zone if no name is given. 

And we can just simply create a new zone with the given name if it doesn't exits already.